include/dali/core/copy_vector_helper.h

```diff
@@ -19,12 +19,44 @@
 #include <utility>
 #include <vector>
 
+#include "dali/core/small_vector.h"
+
 namespace dali {
 
 namespace detail {
 
-template <typename T, typename S>
-void copy_vector(std::vector<T> &out, const std::vector<S> &in) {
+template <typename From, typename To>
+struct is_explicitly_convertible {
+  static void sink_cast_target(To);
+
+  template <typename F, typename T>
+  constexpr static auto test(int) // NOLINT
+      -> decltype(sink_cast_target(static_cast<T>(std::declval<F>())), true) {
+    return true;
+  }
+
+  template <typename F, typename T>
+  constexpr static bool test(...) {
+    return false;
+  }
+
+ public:
+  constexpr static bool value = test<From, To>(0);  // = requires(From t) { static_cast<To>(t); };
+};
+
+template <typename T, typename V1, typename V2>
+inline void copy_vector_impl(V1 &out, const V2 &in);
```

Comment:
nitpick/question: why not simply define the `copy_vector_impl` here?

Reply:
Artifact of previous version. It can even be inlined now. Will do thanks.